runner/query_run.go

```diff
@@ -39,85 +39,74 @@ func SetupQueryRunnerForAllServers(ctx context.Context, servers []*state.Server,
 	}
 }
 
+func runQueryOnDatabase(ctx context.Context, server *state.Server, collectionOpts state.CollectionOpts, logger *util.Logger, id int64, query *state.QueryRun) (string, error) {
+	if query.Type != pganalyze_collector.QueryRunType_EXPLAIN {
+		logger.PrintVerbose("Unhandled query run type %d for %d", query.Type, query.Id)
+		return "", errors.New("Unhandled query run type")
+	}
+
+	db, err := postgres.EstablishConnection(ctx, server, logger, collectionOpts, query.DatabaseName)
+	if err != nil {
+		return "", err
+	}
+	defer db.Close()
+
+	if postgres.StatsHelperExists(ctx, db, "explain_analyze") {
+		logger.PrintVerbose("Found pganalyze.explain_analyze helper function in database \"%s\"", query.DatabaseName)
+	} else {
+		return "", fmt.Errorf("Required helper function pganalyze.explain_analyze is not set up")
+	}
+
+	err = postgres.SetStatementTimeout(ctx, db, 60*1000)
+	if err != nil {
+		return "", err
+	}
+
+	pid := 0
+	err = db.QueryRow(postgres.QueryMarkerSQL + "SELECT pg_backend_pid()").Scan(&pid)
+	if err != nil {
+		return "", err
+	}
+	server.QueryRunsMutex.Lock()
+	server.QueryRuns[id].BackendPid = pid
+	server.QueryRunsMutex.Unlock()
+
+	for name, value := range query.PostgresSettings {
+		_, err = db.ExecContext(ctx, postgres.QueryMarkerSQL+"SET %s = %s", pq.QuoteIdentifier(name), pq.QuoteLiteral(value))
```

Comment:
When testing this locally with `statement_timeout` set to `1s`, this fails with `pq: syntax error at or near "%"`. Looks like a `fmt.Sprintf` call is needed.

Reply:
Oh yeah, strange, I must have dropped that in a refactoring by accident.

Reply:
This now works as expected for your test case (that you had in the other PR).

That said, I do wonder if we should explicitly block changing `statement_timeout` through the parameters in the query run, or at least have some higher limit that's a hard limit? (i.e. can't be changed by the server side)

Maybe something like 1 minute as the default limit, can be raised to up to 10 minutes by the server side, but nothing higher?

Assuming we'd want to do that in the future, its likely easier to enforce by passing it as a separate value (not a free form parameter, i.e block it in `parameters`), so we don't have to deal with `20min` vs `20s` parameter value parsing.

Reply:
Agreed it makes sense to block it here and make it a standalone setting so we can validate it. But applying a max value is going to depend on the type of query run that's being performed; if in the future we're doing a concurrent reindex it should be fine for that to run for an arbitrary amount of time.

Reply:
Yeah, agreed. I'll filter it out here for now and we can add customizing it in a separate PR.

Reply:
Actually on second thought a separate field isn't really needed: we can validate the value in `postgres_settings` since the key value is always going to be named `statement_timeout`.

For now though, we can bypass this issue by making the statement timeout non-configurable. That can be done by moving the existing `postgres.SetStatementTimeout` call after this code, so the hardcoded 1 minute timeout is always used.

Reply:
> Actually on second thought a separate field isn't really needed: we can validate the value in `postgres_settings` since the map key is always going to be named `statement_timeout`.

The problem is not the key, but the value - since you can use different duration units (seconds vs minutes), we'd have to parse them in order to validate the value.

> For now though, we can bypass this issue by making the statement timeout non-configurable. That can be done by moving the existing postgres.SetStatementTimeout call after this code, so the hardcoded 1 minute timeout is always used.

Sure, I'll move it down for now so it takes effect after the other settings.